completed

dreamQA opened the pull request:
sprint_6 - гайдай А.Ф 18а когорта

Commit:
completed

ShamilovR commented on pages/faq_page.py:
```diff
@@ -0,0 +1,32 @@
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from data import Urls
+from conftest import browser
```
Нужно исправить: фикстуры не нужно импортировать в пейджи, они используются только в тестовых методах

ShamilovR commented on pages/faq_page.py:
```diff
@@ -0,0 +1,32 @@
+from conftest import browser
+from locators.faq_locators import FaqLocators
+
+class QestionsPage:
```
Нужно исправить: необходимо реализовать базовый класс и вынести туда все методы для работы с драйвером и элементами. Далее пейджи должны наследоваться от базового класса и переиспользовать его методы. Базовый класс - абстракция, созданная только для наследования.

ShamilovR commented on pages/faq_page.py:
```diff
@@ -0,0 +1,32 @@
+class QestionsPage:
+    @allure.step("Открытие браузера")
+    def open_browser(self,browser):
+        browser.get(Urls.MAIN_PAGE_URL)
```
Нужно исправить: здесь и далее: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

ShamilovR commented on pages/faq_page.py:
```diff
@@ -0,0 +1,32 @@
+    @allure.step("Открытие браузера")
+    def open_browser(self,browser):
+        browser.get(Urls.MAIN_PAGE_URL)
+        return self
```
Можно лучше: возвращать self из метода необязательно. Это может быть актуально при использовании паттерна chain of calls. Однако, он здесь не используется.

ShamilovR commented on tests/test_logo.py:
```diff
@@ -0,0 +1,34 @@
+from conftest import browser
+from pages.logo_page import LogoPage
+
+@pytest.fixture
```
Нужно исправить: фикстуры должны храниться в модуле conftest. Такое решение позволяет остальным тестовым сценариям тоже использовать эти фикстуры без необходимости импорта.

ShamilovR commented on tests/test_logo.py:
```diff
@@ -0,0 +1,34 @@
+import allure
+import pytest
+from conftest import browser
```
Нужно исправить: здесь и далее: фикстуры не нужно импортировать. Если они лежат в conftest то pytest их видит без импорта

Commit:
исправлены замечания:
1.Нужно исправить: фикстуры не нужно импортировать в пейджи, они используются только в тестовых методах
2.Нужно исправить: необходимо реализовать базовый класс и вынести туда все методы для работы с драйвером и элементами. Далее пейджи должны наследоваться от базового класса и переиспользовать его методы. Базовый класс - абстракция, созданная только для наследования.
3.Нужно исправить: здесь и далее: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.
4.Нужно исправить: фикстуры должны храниться в модуле conftest. Такое решение позволяет остальным тестовым сценариям тоже использовать эти фикстуры без необходимости импорта.
5.Нужно исправить: здесь и далее: фикстуры не нужно импортировать. Если они лежат в conftest то pytest их видит без импорта

ShamilovR commented on pages/logo_page.py:
```diff
@@ -0,0 +1,46 @@
+
+    @allure.step("Переключение вкладки")
+    def switching_to_the_tab(self):
+        self.browser.switch_to.window(self.browser.window_handles[1])
```
Нужно исправить: здесь и далее: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

Commit:
completed
